packages/protocol/src/client-server-protocol/jsonrpc/base-jsonrpc-glsp-client.ts

```diff
@@ -91,6 +96,9 @@ export class BaseJsonrpcGLSPClient implements GLSPClient {
         try {
             this.state = ClientState.Starting;
             const connection = await this.resolveConnection();
+            connection.onNotification(JsonrpcGLSPClient.ActionMessageNotification, msg =>
```

Comment:
nit: Should we add that to the `doCreateConnection` method where we already add handlers for `onError` and `onClose`? I think it would be a better fit.